delta/middleman.py

```diff
@@ -22,26 +20,40 @@
 import timeit
 import time
 
-from streaming.reader_nompi import reader_gen
-from streaming.writer_nompi import writer_gen
-from analysis.channels import channel_range
-
+from sources.dataloader import get_loader
+from streaming.writers import writer_gen
+from streaming.reader_mpi import reader_gen
+from data_models.helpers import gen_channel_name, gen_var_name
 
 @attr.s
 class AdiosMessage:
     """Storage class used to transfer data from Kstar(Dataman) to local PoolExecuto."""
     tstep_idx = attr.ib(repr=True)
+    # (2020/11) jyc: Good place to wrap numpy data with ecei_chunk?
```

Comment:
It may be good to pack the raw numpy data into the same data-type in both, processor and middleman. Then the logic in both reads identical. You can try instantiating an ecei_chunk here, passing tb=None in the constructor.